From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: cpw <cpw@weeksfamily.ca>
Date: Mon, 12 Oct 2015 14:44:39 -0400
Subject: [PATCH] Fix IncludeEntireClasspath so it doesn't dump out random
 other files from the classpath. Also, it doesn't try and save undecompiled
 additional classes too.


diff --git a/src/org/jetbrains/java/decompiler/main/Fernflower.java b/src/org/jetbrains/java/decompiler/main/Fernflower.java
index 65733ac2c8ea53277e5a2ffe54ea7e91e0c6fc97..6afe975fe2034ad25b7810a0bdc59ebcff8453e1 100644
--- a/src/org/jetbrains/java/decompiler/main/Fernflower.java
+++ b/src/org/jetbrains/java/decompiler/main/Fernflower.java
@@ -113,7 +113,8 @@ public class Fernflower implements IDecompiledData {
         if (found.contains(file.getAbsolutePath()))
           continue;
 
-        if (file.exists()) {
+        // only add .class files from classpath
+        if (file.exists() && (file.getName().endsWith(".class") || file.getName().endsWith(".jar"))) {
           DecompilerContext.getLogger().writeMessage("Adding File to context from classpath: " + file, Severity.INFO);
           structContext.addSpace(file, false);
           found.add(file.getAbsolutePath());
diff --git a/src/org/jetbrains/java/decompiler/struct/ContextUnit.java b/src/org/jetbrains/java/decompiler/struct/ContextUnit.java
index 850c3d84888d95343abd8a3c297baac58783a8c3..e3043962de86e4823458766ff2afda4eb3e71e5f 100644
--- a/src/org/jetbrains/java/decompiler/struct/ContextUnit.java
+++ b/src/org/jetbrains/java/decompiler/struct/ContextUnit.java
@@ -110,6 +110,7 @@ public class ContextUnit {
         // classes
         for (int i = 0; i < classes.size(); i++) {
           StructClass cl = classes.get(i);
+          if (!cl.isOwn()) continue;
           String entryName = decompiledData.getClassEntryName(cl, classEntries.get(i));
           if (entryName != null) {
             String content = decompiledData.getClassContent(cl);
